import/popups/ContextMenu.qml

```diff
@@ -0,0 +1,269 @@
+
+    visible: opacity > 0.0
+    opacity: 0.0
+
+    width: parent.width
+    height: parent.height
+    gradient: Gradient {
+        GradientStop { position: 0.0; color: Theme.highlightDimmerColor }
+        GradientStop { position: 1.0; color: Theme.rgba(Theme.highlightDimmerColor, .91) }
+    }
+
+    Behavior on opacity { FadeAnimation { duration: 300 } }
+
+    Column {
+        width: parent.width
+        spacing: Theme.paddingMedium
+        anchors.top: parent.top
+        anchors.topMargin: Theme.paddingLarge*2
+
+        Label {
+            id: title
+            anchors.horizontalCenter: parent.horizontalCenter
+            visible: root.linkTitle.length > 0
+            text: root.linkTitle
+            width: root.width - Theme.paddingLarge*2
+            elide: Text.ElideRight
+            wrapMode: Text.Wrap
+            maximumLineCount: 2
+            color: Theme.highlightColor
+            font.pixelSize: Theme.fontSizeExtraLarge
+            horizontalAlignment: Text.AlignHCenter
+            opacity: .6
+        }
+
+        Label {
+            anchors.horizontalCenter: parent.horizontalCenter
+            color: Theme.highlightColor
+            text: root.imageSrc.length > 0 ? root.imageSrc : root.url
+            width: root.width - Theme.paddingLarge*2
+            wrapMode: Text.Wrap
+            elide: Text.ElideRight
+            maximumLineCount: landscape ? 1 : 4
+            font.pixelSize: title.visible ? Theme.fontSizeMedium : Theme.fontSizeExtraLarge
+            horizontalAlignment: Text.AlignHCenter
+            opacity: .6
+        }
+    }
+
+    Column {
+        id: menu
+
+        property Item highlightedItem
+
+        anchors.bottom: parent.bottom
+        anchors.bottomMargin: landscape ? Theme.paddingLarge : Theme.itemSizeSmall
+        width: parent.width
+
+        MenuItem {
+            visible: !isHyperTextLink && !isImage
+            text: {
+                if (isMailto) {
+                    //: Write email
+                    //% "Write email"
+                    return qsTrId("sailfish_components_webview_popups-me-write-email")
+                } else if (isTel) {
+                    //: Call, context of calling via voice call
+                    //% "Call"
+                    return qsTrId("sailfish_components_webview_popups-me-call")
+                } else if (isSMS) {
+                    //: Send message (sms)
+                    //% "Send message"
+                    return qsTrId("sailfish_components_webview_popups-me-send-message")
+                } else {
+                    //: Open link in current tab
+                    //% "Open link"
+                    return qsTrId("sailfish_components_webview_popups-me-open_link")
+                }
+            }
+
+            onClicked: {
+                root._hide()
+                Qt.openUrlExternally(root.linkHref)
+            }
+        }
+
+        MenuItem {
+            visible: root.isNavigable
```

Comment:
Let's add explicit tabModel property for the ContextMenu and hide "open x in a new tab" context menu options if tabModel is not present.